### PR TITLE
TotalHits for Lucene Queries.

### PR DESCRIPTION
**Pull request by Skrypt**
Add the ability to return a count of possible results with a Lucene Query. The count is returned with the Lucene Query directly without requiring to Query a second time to get possible results count.

Previously to achieve this you could use a Lucene Query that was using a (Size = really high number) to test it. I did not liked it so I managed to return the Count of results from a Lucene Query without affecting much the query perf. The perf is actually no different with or without it.

screencapture-localhost-44300-OrchardCore-Lucene-Admin-Query-2019-10-17-17_06_05

### CODE REVIEW

File: src/OrchardCore/OrchardCore.Queries.Abstractions/IQueryResult.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+namespace OrchardCore.Queries
+{
+    public interface IQueryResult
+    {
+        public object Items { get; set; }
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Queries.Abstractions/IQueryResult.cs]: Maybe an `IEnumerable<object>` as the results used to be that

File: src/OrchardCore/OrchardCore.Queries.Abstractions/IQueryResult.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+namespace OrchardCore.Queries
+{
+    public interface IQueryResult
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Queries.Abstractions/IQueryResult.cs]: `IQueryResults`

File: src/OrchardCore/OrchardCore.Queries.Abstractions/IQueryResult.cs (hunk 1 above)
Comment (Skrypt) [reply src/OrchardCore/OrchardCore.Queries.Abstractions/IQueryResult.cs]: A SQL Query could return a single value. But I will try.

Comment (Skrypt) [reply src/OrchardCore/OrchardCore.Queries.Abstractions/IQueryResult.cs]: image
 Works

Comment (Skrypt) [reply src/OrchardCore/OrchardCore.Queries.Abstractions/IQueryResult.cs]: Though I broke something with when we check the option "return documents"

File: src/OrchardCore/OrchardCore.Queries.Abstractions/LuceneQueryResult.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+using System.Collections.Generic;
+
+namespace OrchardCore.Queries
+{
+    public class LuceneQueryResult: IQueryResults
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Queries.Abstractions/LuceneQueryResult.cs]: `LuceneQueryResults`

File: src/OrchardCore/OrchardCore.Queries.Abstractions/SQLQueryResult.cs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+using System.Collections.Generic;
+
+namespace OrchardCore.Queries
+{
+    public class SQLQueryResult : IQueryResults
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Queries.Abstractions/SQLQueryResult.cs]: `SQLQueryResults`

**Comment by sebastienros** [comment]
I approved, merge when your tests are ok

**Comment by Skrypt** [comment]
We are missing unit test for SQL Queries that returns documents. I think the same goes with Lucene Queries.

**Comment by hyzx86** [comment]
@Skrypt 
Is there any way to return the count information when using the rest API?
image

**Comment by Skrypt** [comment]
Open an issue about it. I don't think it will work as of now but it should.

**Pull request merged**